### ISSUE TITLE
Problem with movie versions

### ISSUE THREAD

**Issue by didaktiik** [jellyfin/jellyfin-androidtv#1531]
### Describe the bug

Hello,

I been having this issue with versions ever since they had been introduced to Android TV beta versions (beta4?). As you can see bellow, my files follow the name convention required for movies versions.

- 'Home' version stands for LAN streaming (best quality), basically for home use in general
- 'Internet' version is a version made for remote streaming (bitrate is restricted to 4-8Mbit, Level 4.1, AAC & AC3 sound)

For the most part of my movie library, Android TV client (Nvidia Shield 2017,  Jellyfin client 0.13.0 is installed) chooses the 'Home' version at Home, but for some movies it insists on 'Internet' version not matter what I do. Not only the 'Internet' version is pre-seleted by default (in version selector) but it plays that version as well despite the 'Home' version plays fine, when chosen manually (direct play, no transcode or some compatibility issue).

Remote streaming (from a remote location, using different Nvidia Shield TV, but same client version) always chooses the 'Internet' version, most likely due the max bitrate settings set in the client (10Mbit/s).

Example of a movie (folder listing, so you can see file naming) where an incorrect version is selected by default:

```
[thatuser@manjaro-pc Home Alone 2: Lost in New York (1992)]$ pwd
/media/NAS414/Plex/Movies/Home Alone 2: Lost in New York (1992)
[thatuser@manjaro-pc Home Alone 2: Lost in New York (1992)]$ ls -la
total 29121192
drwxr-xr-x   2 nobody users        4096 Mar  8 15:15  .
drwxrwxrwx 126 nobody users        4096 Mar 17 22:06  ..
-rwxrw-rw-   1 nobody users       66222 Oct 13  2020 'Home Alone 2: Lost in New York (1992) - Home.cs.srt'
-rwxrwxrwx   1 nobody users       91334 Oct 13  2020 'Home Alone 2: Lost in New York (1992) - Home.en.srt'
-rwxrwxrwx   1 nobody users 25332523689 Dec 31 11:06 'Home Alone 2: Lost in New York (1992) - Home.mkv'
-rw-rw-rw-   1 nobody users       66222 Oct 13  2020 'Home Alone 2: Lost in New York (1992) - Internet.cs.srt'
-rw-rw-rw-   1 nobody users       91334 Oct 13  2020 'Home Alone 2: Lost in New York (1992) - Internet.en.srt'
-rw-r--r--   1 nobody users  4487228001 Feb 21 12:07 'Home Alone 2: Lost in New York (1992) - Internet.mp4'
```

### Application version

0.13.0

### Where did you install the app from?

Google Play

### Device information

Nvidia Shield 2017

### Android version

Android 11

### Jellyfin server version

10.7.7

**Comment by sourpatched** [comment]
Have you looked at https://jellyfin.org/docs/general/server/media/movies.html 
"To distinguish between versions, each filename needs to have a space, hyphen, space, and then a label. Labels are not predetermined and can be made up by the user."

**Comment by didaktiik** [comment]
I have. See my filenames bellow:

Home Alone 2: Lost in New York (1992) - Home.mkv
Home Alone 2: Lost in New York (1992) - Internet.mp4

space - hyphen - space - label (Home, Internet)

**Comment by sourpatched** [comment]
I wonder if this is happening because it's MP4 file?

**Comment by didaktiik** [comment]
No, the vast majority of my movie library is like that - Home.mkv and Internet.mp4 and it works. For some movies it doesn't. There is no pattern, that's why it is quite baffling to say the least.

**Comment by didaktiik** [comment]
I tried to rename mp4 to mkv (it plays fine so the extension doesn't really matter), and that did the trick for some movies. Still, many have refused. There is no clear pattern, no rhyme or reason when deciding between those two "profiles". As wiki says, the deciding factor is an alphabetical order and the factor is satisfied:

- Home (comes first)
- Internet (comes after the first).

**Comment by sourpatched** [comment]
"As wiki says, the deciding factor is an alphabetical order and the factor is satisfied"

Could be that it isn't hard coded in the app.

**Comment by didaktiik** [comment]
To investigate further, I have installed another client - Jellycon addon for Kodi. Suprisingly enough, it does the same thing as Jellyfin for Android TV- for some movies (the same ones!) it prefers "Internet" to "Home" profile.

Maybe this rules out the client out of quation and the server is the one to blame? I don't know. But wiki is clearly not accurate it seems.

If there was another deciding factor I wasn't able to figure it out, and I have tried all possible combinations. As your suggested, for some movies it helped to rename mp4 to mkv. For some , it didn't.

**Comment by sourpatched** [comment]
Can you post both media details? you can copy them from the web GUI if you have it installed.

**Comment by didaktiik** [comment]
Media info copied from Jellyfin web looks like garbage (formatting is completely lost), as you can see in the 3rd attachement, so I included the output of Mediainfo application

[Media info Jellyfin Web.txt](https://github.com/jellyfin/jellyfin-androidtv/files/8401862/Media.info.Jellyfin.Web.txt)
.[Jumanji (1995) - Home.mkv.txt](https://github.com/jellyfin/jellyfin-androidtv/files/8401857/Jumanji.1995.-.Home.mkv.txt)
[Jumanji (1995) - Internet.mkv.txt](https://github.com/jellyfin/jellyfin-androidtv/files/8401858/Jumanji.1995.-.Internet.mkv.txt)

**Comment by didaktiik** [comment]
directory listing so you can see filenames

`-rw-r--r--   1 nobody users       62868 Feb  9 09:25 'Jumanji (1995) - Home.cs.srt'
-rw-r--r--   1 nobody users 24341509828 Apr  1 21:00 'Jumanji (1995) - Home.mkv'
-rw-r--r--   1 nobody users       62868 Feb  9 09:25 'Jumanji (1995) - Internet.cs.srt'
-rw-r--r--   1 nobody users  3990622936 Apr  1 21:02 'Jumanji (1995) - Internet.mkv'
`

**Comment by didaktiik** [comment]
Directory listing
```

-rw-r--r--   1 nobody users       62868 Feb  9 09:25 'Jumanji (1995) - Home.cs.srt'
-rw-r--r--   1 nobody users 24341509828 Apr  1 21:00 'Jumanji (1995) - Home.mkv'
-rw-r--r--   1 nobody users       62868 Feb  9 09:25 'Jumanji (1995) - Internet.cs.srt'
-rw-r--r--   1 nobody users  3990622936 Apr  1 21:02 'Jumanji (1995) - Internet.mkv'
```

**Comment by sourpatched** [comment]
https://jellyfin.org/docs/general/server/media/movies.html#order-of-versions
Movie versions are presented in an alphabetically sorted list. An exception applies to resolution names, which are sorted in descending order from highest to lowest resolution. A version name qualifies as a resolution name when ending with either a p or an i.

What happens if you rename them to 1080p.mkv and 720p.mkv (Jumanji (1995) - 1080p.mkv) and then scan for changes.

**Comment by didaktiik** [comment]
It worked, for this movie.

However I did rename more movies, particulary Back to the Future trilogy:

Back to the Future (1985) - 1080p.mkv
Back to the Future (1985) - 720p.mkv

Back to the Future 2 (1989) - 1080p.mkv
Back to the Future 2 (1989) - 720p.mkv

Back to the Future 3 (1990) - 1080p.mkv
Back to the Future 3 (1990) - 720p.mkv

The result is inconclusive:

Back to the Future (1985), 1080p is selected by default.
Back to the Future 2 (1989), 720p is selected by default.
Back to the Future 3 (1990), 720p is selected by default.

-- 
  
  ***@***.***

**Comment by sourpatched** [comment]
> 
I would try and change the P to an I and refresh, I don't know if that will fix it but it seems like a server issues?

**Comment by didaktiik** [comment]
It looks like a server related issue, yes.

I am not sure what else to do, to let developers know about the issue, which is obviously there.

-- 
  
  ***@***.***

**Comment by sourpatched** [comment]
@nielsvanvelzen  or @mueslimak3r can you please move this to Jellyfin server?

You could give 10.8 beta a try, I haven't had any server issues so far and the app works okay with it.
There is also a media player for all OS.

**Comment by didaktiik** [comment]
I primarily use Jefllyfin on Android TV (Nvidia Shield TV). As far as I remember Jellyfin Web and iOS clients (iPad and iPhone) always choose the "Home" profile (or whatever is called, 1080p etc - the best quality profile in general) regardless of max bitrate allowed for playback client-wise.

I will rather wait a while before moving to beta server. On a client side (GoolePlay), I signed up to a beta program :-)
